scipy/optimize/_minimize.py

```diff
@@ -281,6 +273,17 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
 
     **Bound-Constrained minimization**
 
+    Method :ref:`Powell <optimize.minimize-powell>` is a modification
+    of Powell's method [3]_, [4]_ which is a conjugate direction
+    method. It performs sequential one-dimensional minimizations along
+    each vector of the directions set (`direc` field in `options` and
+    `info`), which is updated at each iteration of the main
+    minimization loop. The function need not be differentiable, and no
+    derivatives are taken. If bounds are provided, they will be
+    respected at every iteration, and a bounded line search method will
+    be used. If bounds are not provided, then an unbounded line search
+    method will be used.
+
```

Comment:
✓

Reply:
BTW is there a reference for this bounded line search, or is this a common modification to unbounded line search optimization methods?

Reply:
To be honest, I am not sure if this is a common modification or not. I thought about it a little while ago and started using it with some of my research and found it to give me very good results for some of the problems I was working on. I am not sure of any references. For many of the problems I am interested in, I found this bounded Powell modification to outperform all other minimizers implemented in SciPy. Similarly, for many problems, I found it to perform worse than other minimizers.

Reply:
Gotcha. @antonior92 have you seen this sort of thing before? 

SciPy's scope is usually limited to standard algorithms. I wonder if a useful but nonstandard extension to a standard algorithm is OK.

Is there any reason that the same idea wouldn't apply to other line search optimizers? If this is not standard, I would guess there are reasons not to do it, as the idea is pretty natural.

Reply:
I haven't seem this before, but it does seems quite a natural way of dealing with bounds in line-search methods...

I agree that it is still within the scope of SciPy, since the modication, while not entirely standard is still a quite natural addition to improve the performance for box constraints...

Reply:
Agreed :)

Reply:
Is there anything else you guys think I need to do?

Reply:
@rgommers OK to include this if it's a useful extension and looks very reasonable but is not really standard?

Reply:
I'd say if it works better than the old code, why not. This is not like including a whole new algorithm, it's an improvement on what we already have - so no need to decide based on whether this is a published peer-reviewed idea.

No comment on the actual PR, I'll leave that up to the experts here.